### PR TITLE
Add batching for metric upload

### PR DESCRIPTION
**Pull request by k0nserv**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -307,6 +399,10 @@ impl DDStatsClient {
         interval: Option<f64>,
     ) -> Tagger<'_> {
         let mangled = format!("{}.{}", self.namespace, mangle_safe(name));
+        assert!(
```
Comment (k0nserv) [inline src/lib.rs:402]: This is in preparation for V2 which has this requirement

File: src/lib.rs [hunk 2]
```diff
@@ -422,47 +528,48 @@ impl DDStatsClient {
...
-            );
-            let body = serde_json::to_string(&Series::new(payload.metrics.iter()))
+        // Metrics
+
+        use std::ops::Deref;
+        let batches = batch_metrics(
+            payload.metrics.into_iter().map(Cow::Owned),
+            DATADOG_V1_METRIC_MAX_SIZE,
+        );
+        let api_url = format!("http://httpbin/post?api_key={}", api_key);
```
Comment (k0nserv) [inline src/lib.rs:538]: This needs to be fixed, left over from when I was testing 😅

**Review by lolgesten — APPROVED** [review]
Some warnings to fix, but overall looks fine.

**Comment by k0nserv** [comment]
I believe the warnings are all unrelated to these changes, but I'll fix them on `main`

**Pull request merged**